src/Adapter/Address/CommandHandler/BulkDeleteAddressHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Address\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\Address\Command\BulkDeleteAddressCommand;
+use PrestaShop\PrestaShop\Core\Domain\Address\CommandHandler\BulkDeleteAddressHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Address\Exception\DeleteAddressException;
+
+/**
+ * Handles command which deletes addresses in bulk action
+ */
+final class BulkDeleteAddressHandler extends AbstractAddressCommandHandler implements BulkDeleteAddressHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(BulkDeleteAddressCommand $command)
+    {
+        foreach ($command->getAdressIds() as $addressId) {
+            $addressIdValue = $addressId->getValue();
+
+            if (!$this->deleteLegacyAddress($addressId)) {
```

Comment:
Seems like `deleteLegacyAddress` function handles two responsibilities - first it checks if address exists and throws exception if not and then just later it actually performs delete operation. I think its better to split this behavior

Reply:
You both offering to create and validate legacy object in each handler and accept it as argument in this method?

Reply:
I wouldn't call asserting "responsibility" in this case. It's more like integrity check and what is more, it is extracted into separate reusable method. 

However, it may make sense to accept `Manufacturer` as method argument and avoid creating it here at all.